optimizer/lib/transformers/AddBlurryImagePlaceholders.js

```diff
@@ -155,6 +162,49 @@ class AddBlurryImagePlaceholders {
     });
     return false;
   }
+
+  /**
+   * Checks if an image should have a blurred image placeholder.
+   * @param {Node} node The DOM element that is being checked to see if it
+   * should have a blurred placeholder.
+   * @param {string} src The image source that is being checked.
+   * @param {string} tagName The type of element that is being checked.
+   * @return {boolean} Whether or not the element should have a blurred
+   * placeholder child.
+   * @private
+   */
+  passesHueristic_(node, src, tagName) {
+    // Checks to see if the image is a jpeg.
+    if (!src.endsWith('.jpg') && !src.endsWith('jpeg')) {
+      return false;
+    }
+
+    // Checks if the image is a poster or a responsive image
+    const isPoster = tagName == 'amp-video';
+    const isResponsiveImg = (tagName == 'amp-img' &&
+      node.attribs.layout == 'responsive');
+    if (isPoster || isResponsiveImg) {
+      // Ensures image is not a child of a template
+      return !this.templateAncestor_(node);
```

Comment:
A more efficient approach is to skip templates when iterating over the DOM nodes. Similar to here: https://github.com/ampproject/amp-toolbox/blob/master/optimizer/lib/transformers/PreloadImages.js#L53

Reply:
It might make sense to extract this into a helper method in [HTMLDomHelper](https://github.com/ampproject/amp-toolbox/blob/master/optimizer/lib/HtmlDomHelper.js) as it's not quite trivial because of white space handling.